packages/headless/src/controllers/commerce/recommendations/headless-recommendations.ts

```diff
@@ -0,0 +1,118 @@
+import {Schema} from '@coveo/bueno';
+import {createSelector} from '@reduxjs/toolkit';
+import {CommerceAPIErrorStatusResponse} from '../../../api/commerce/commerce-api-error-response';
+import {Product} from '../../../api/commerce/common/product';
+import {
+  CommerceEngine,
+  CommerceEngineState,
+} from '../../../app/commerce-engine/commerce-engine';
+import {configuration} from '../../../app/common-reducers';
+import {contextReducer as commerceContext} from '../../../features/commerce/context/context-slice';
+import {recommendationSlotDefinition} from '../../../features/commerce/recommendation/recommendation';
+import {
+  fetchRecommendations,
+  updateRecommendationSlotId,
+} from '../../../features/commerce/recommendation/recommendation-actions';
+import {recommendationV2Reducer as recommendation} from '../../../features/commerce/recommendation/recommendation-slice';
+import {loadReducerError} from '../../../utils/errors';
+import {validateInitialState} from '../../../utils/validate-payload';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+/**
+ * The `Recommendations` controller exposes a method for retrieving recomendations content in a commerce interface.
+ */
+export interface Recommendation extends Controller {
+  /**
+   * Fetches the recommendations.
+   */
+  refresh(): void;
+
+  /**
+   * A scoped and simplified part of the headless state that is relevant to the `Recommendation` controller.
+   */
+  state: RecommendationState;
+}
+
+export interface RecommendationState {
+  headline: string;
+  products: Product[];
```

Comment:
@Spuffynism should I use this type or `ProductRecommendation`?

Reply:
`Product` is the right one, since it's the type for products as returned from the commerce api.